crates/tbdex/src/resources/mod.rs

```diff
@@ -36,10 +37,21 @@ impl fmt::Display for ResourceKind {
     }
 }
 
+static COUNTER: AtomicU64 = AtomicU64::new(0);
+
 impl ResourceKind {
     pub fn typesafe_id(&self) -> Result<String> {
         let dynamic_type = DynamicType::new(&self.to_string())?;
-        Ok(TypeSafeId::from_type_and_uuid(dynamic_type, Uuid::new_v4()).to_string())
+        let timestamp_nanos = Utc::now().timestamp_nanos_opt().unwrap();
```

Comment:
Can we map the error into a TbdexError instead of risking a panic from the unwrap?